### PR TITLE
Update to latest mirage libraries, drop deprecated Lwt_log

### PR DESCRIPTION
**Pull request by psafont**
Also drops travis for github actions.

I'm not familiar with all the mirage modules and packages, let me know if the metadata about them can be more precise.

Uloading coverage to coveralls is failing because of missing $PULL_REQUEST_NUMBER, I think it's transitory because the master branch doesn't have CI activated yet and it's technically running under the fork: https://github.com/psafont/shared-block-ring/actions/runs/832296254

### CODE REVIEW

**Comment by avsm** [comment]
could activate ocaml-ci on this as well...

**Comment by avsm** [comment]
`opam package-upgrade` should do the trick @psafont

File: dune-project [hunk 1]
```diff
@@ -1,2 +1,5 @@
-(lang dune 1.0)
+(lang dune 2.7)
```
[the remark this answers is not in the document]
Comment (undu) [reply dune-project:1]: instrumenting integration for bisect_ppx

**Comment by psafont** [comment]
I have changes pending to change the logging module and change the deprecated lwt_log with logs, but changing its signature means dealing with the logging in message-switch, which isn't trivial so I'd rather do it on another PR

**Comment by psafont** [comment]
I've fixed the build in old ocaml versions and dropped the use of the deprecated Lwt_log.

File: shared-block-ring.opam [hunk 2]
```diff
@@ -1,38 +1,43 @@
-opam-version: "1.2"
+opam-version: "2.0"
+synopsis: "A single-consumer single-producer queue on a block device"
+description: """
+This is a simple queue containing variable-length items stored on a
+          disk, in the style of Xen shared-memory-ring."""
 maintainer: "jonathan.ludlam@citrix.com"
-authors: [ "David Scott" "Jon Ludlam" "Si Beaumont" ]
+authors: ["David Scott" "Jon Ludlam" "Si Beaumont"]
```
Comment (djs55) [inline shared-block-ring.opam:7]: BTW feel free to add yourself to the list of `authors`!

Comment (avsm) [inline shared-block-ring.opam:7]:
```suggestion
authors: ["David Scott" "Jon Ludlam" "Si Beaumont" "Pau Ruiz Safont"]
```

**Pull request merged**